### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by waelelsafty07**
### Interface

   - Create a class Nameable.
       - Implement a method called correct_name that will raise a NotImplementedError.
   
### Turn your Person class to Nameable

   - Make sure that your Person class inherits from Nameable
   - Make sure that this class has a method correct_name implemented. It should simply return the name attribute.

### Prepare base Decorator
   - Make sure that it inherits from Nameable.
   - In the constructor assign a nameable object from params to an instance variable.
   - Implement the correct_name method that returns the result of the correct_name method of the @nameable.

### Prepare CapitalizeDecorator and TrimmerDecorator

- For the CapitalizeDecorator:
  - Create a class that inherits from the base Decorator class.
  - Implement a method correct_name that capitalizes the output of @nameable.correct_name.
- For the TrimmerDecorator:
  - Create a class that inherits from the base Decorator class.
  - Implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: decorator.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+require_relative 'nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  attr_accessor :nameable
```
Comment (MahmoudBakr23) [inline decorator.rb:9]: **Optional**
- Please keep in mind that the `attr_accessor` is an association-generator between two classes to identify the relationship type between them. So, it's always common to place it before the initialization. More logical right?

File: person.rb [hunk 2]
```diff
@@ -1,5 +1,7 @@
-class Person
+require_relative 'nameable'
+class Person < Nameable
   def initialize(age, name = 'unknown', parent_permission: true)
+    super()
     @id = Random.rand(1..188)
     @name = name
     @age = age
```
Comment (MahmoudBakr23) [inline person.rb]: - [ ] After you imported the `nameable` using `require_relative`, it's important to add the `correct_name` method here as well to return the passed `name`. Please add the method correctly in the `Person` class. 🙏

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @waelelsafty07 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Great work so far working with the decorator classes and their custom methods
- Descriptive README and PR
- Correct Linters

However, there is always room for improvement.

> I'd like to remind you that your long hours of work are appreciated. However, the code review process is for improving your code, not criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

_Please see the comments under the review._

Cheers, and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ichala — APPROVED** [review]
Hi Student  ,
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  
 
68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f6c3366516333614356304871707a5763552f67697068792e676966

# To Highlight 💪🏻

- Changes Implemented ✔️ 


 Good job implementing the requirements  :+1: Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Optional Suggestions  :

- N/A


**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag  ichala in your question so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**